core/sql/generator/GenPreCode.cpp

```diff
@@ -4934,8 +4934,9 @@ RelExpr * HbaseDelete::preCodeGen(Generator * generator,
   else
   if (isUnique)
     {
-      // do not cancel unique queries.
-      generator->setMayNotCancel(TRUE);
+      if (!generator->oltOptInfo()->multipleRowsReturned())
+	// do not cancel unique queries.
+	generator->setMayNotCancel(TRUE);
```

Comment:
Done updating the comment. Also added a similar check to HbaseUpdate . Hbase Access already had the check in place and added just a comment. I did not find any other place that needed this check.